### PR TITLE
Fixes #20660 - improve proxy features list

### PR DESCRIPTION
**Pull request by ares**
before
before1

after
after1

@Rohoover this is based on your suggestion, any comments?

### CODE REVIEW

**Comment by sean797** [comment]
TBH I quite like the old way but if we want to list them as a list when maybe we should look at multiple columns? 

I think the way this PR currently is it introduces a lot of empty/dead space 👎

**Comment by ares** [comment]
I tend to agree, multiple columns though might not help readability for features with various length. Let's see what @Rohoover and others think. Also we might sort it alphabetically while we're touching it?

**Comment by timogoebel** [comment]
I do prefer the "old" way as well. Mostly for the same reasons as @sean797.

👎 from me.

**Comment by Rohoover** [comment]
I would recommend alphabetical in multiple columns. Traditionally, when looking for specific items, lists are easier to scan (think Table of Contents, Grocery List, etc.). When they are listed with commas, you are sort of forced to scan the entire word in order to search for the next word. In a list, you can scan and stop (maybe even at the first letter) and move on if it wasn't what you were looking for.

**Review by dLobatog — CHANGES_REQUESTED** [review]
Setting to WoC as per @Rohoover 's comment, I also think it doesn't look better in the original screenshot. The refresh button changed from a icon to a button with text, is that intentional?

**Comment by ares** [comment]
Ok, I'll see what I can do with the multi columns list

> The refresh button changed from a icon to a button with text, is that intentional?

Yes, I combined these two to avoid rebasing, that was another recommendation of @Rohoover

**Comment by ares** [comment]
new version
after1

@sean797 @timogoebel @Rohoover @dLobatog what do you think?

**Comment by Rohoover** [comment]
@ares 
Can we keep the button always as an element on it's own row beneath the bulleted columns?

I like the multiple columns -- uses the space better.

**Comment by ares** [comment]
@Rohoover like this?
after2

**Comment by Rohoover** [comment]
bitmoji

File: app/views/smart_proxies/show.html.erb [hunk 1]
```diff
@@ -52,10 +52,27 @@
...
             <%= _('Active features') %>
           </div>
+
+
+            <% @smart_proxy.features.order(:name).pluck(:name).in_groups_of(5, false) do |features| %>
+              <div class="col-md-2">
+                <ul class="">
+                  <% features.each do |feature| %>
+                    <li class=""><%= h(feature).gsub(' ', '&nbsp;').html_safe %></li>
+                <% end %>
```
Comment (sean797) [inline app/views/smart_proxies/show.html.erb]: Indentation is off here

File: app/views/smart_proxies/show.html.erb [hunk 2]
```diff
@@ -52,10 +52,27 @@
...
+
+        <div class="row">
+          <div class="col-md-4"></div>
           <div class="col-md-8">
-            <%= @smart_proxy.features.map(&:name).to_sentence %> <%= refresh_proxy_icon(@smart_proxy, authorizer) %>
+            <%= refresh_proxy_button(@smart_proxy, authorizer) %>
           </div>
         </div>
+
+        <br>
```
Comment (sean797) [inline app/views/smart_proxies/show.html.erb]: Whats this for?

**Review by sean797 — CHANGES_REQUESTED** [review]

**Comment by sean797** [comment]
2 small comments. Looks much better! 👍

File: app/views/smart_proxies/show.html.erb (hunk 2 above)
Comment (ares) [reply app/views/smart_proxies/show.html.erb]: left over from various versions, thanks!

**Comment by ares** [comment]
fixed both, thanks @sean797

**Review by sean797 — APPROVED** [review]
thanks @ares

**Comment by timogoebel** [comment]
I think this looks a little squeezed. The new refresh features button needs more space to the number of managed hosts.

image

**Review by dLobatog — APPROVED** [review]

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ares Setting to WoC based on @timogoebel 's comment, take a look at that if you can - sounds like some padding is required or similar

File: app/views/smart_proxies/show.html.erb (hunk 2 above)
Comment (ares) [reply app/views/smart_proxies/show.html.erb]: so the `<br>` was actually necessary for what @timogoebel reported below

**Comment by ares** [comment]
I reverted to new line, there's now more space again like in the last of my screenshots, any more comments?

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ares & everybody else involved! 😄

**Pull request merged**